sdk/test/metrics/async_metric_storage_test.cc

```diff
@@ -0,0 +1,41 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef ENABLE_METRICS_PREVIEW
+#  include "opentelemetry/sdk/metrics/state/async_metric_storage.h"
+#  include "opentelemetry/common/key_value_iterable_view.h"
+#  include "opentelemetry/sdk/metrics/instruments.h"
+#  include "opentelemetry/sdk/metrics/observer_result.h"
+
+#  include "opentelemetry/sdk/instrumentationlibrary/instrumentation_library.h"
+#  include "opentelemetry/sdk/resource/resource.h"
+
+#  include <gtest/gtest.h>
+#  include <map>
+
+using namespace opentelemetry::sdk::metrics;
+using namespace opentelemetry::sdk::instrumentationlibrary;
+using namespace opentelemetry::sdk::resource;
+
+void measurement_fetch(opentelemetry::metrics::ObserverResult<long> &observer_result)
+{
+  observer_result.Observe(20l);
+  observer_result.Observe(10l);
+}
+
+TEST(AsyncMetricStorageTest, BasicTests)
+{
+  auto metric_callback            = [](MetricData &metric_data) { return true; };
+  InstrumentDescriptor instr_desc = {"name", "desc", "1unit", InstrumentType::kCounter,
+                                     InstrumentValueType::kLong};
+  auto instrumentation_library    = InstrumentationLibrary::Create("instr_lib");
+  auto resource                   = Resource::Create({});
+  MetricCollector *collector;
```

Comment:
are we using null `collector` in purpose?

Reply:
We still don't have  MetricCollector interface defined, and also AsyncMetricStorage::Collect() is not implemented. This will be fixed once the metric collection is implemented.